src/rust/sdk/src/errors.rs

```diff
@@ -44,6 +112,12 @@ pub enum EconiaError {
     #[error("failed submitting aptos transaction")]
     FailedSubmittingTransaction,
 
+    #[error("invalid response from the contract")]
+    InvalidResponse,
```

Comment:
IDK, if using an older SDK with a newer version of the contract (or the other way around), you could have an invalid response (if the response structure changes). I thought this could be the enum variant for that.

Reply:
It's no big deal, however, it's impossible for future versions of the contract to break this code due to the Compatible Upgrade Policy which means fields can't be removed from structs for example, and entry function signatures can't change.

Reply:
I think that `InvalidResponse` is fine, as it looks like this error is getting used for view functions

The Aptos compatibility guarantees are weak in the sense that they only check ABI modifications, so it is possible to send back a new data schema that technically doesn't break the ABI but is semantically invalid nonetheless

Like if a field is a vec of `u8` that is supposed to represent a hash digest, but the field is deprecated and the field just comes back empty